### PR TITLE
Add anonymise database task

### PR DESCRIPTION
**Pull request by WithoutPants**
Occasionally, it is necessary to obtain a user's stash database for debugging issues. This PR provides a way for a user to share their database without leaking information they consider private.

Adds `Anonymise`, and `Download anonymised` tasks to the Tasks page - similar to the backup tasks.

Anonymise makes a copy of the database, randomising all sensitive data:
- deletes all image blobs
- renames all folders to be equal to their id numbers
- renames all files to be equal to their id numbers
- replaces fingerprint values with random hex numbers
- replaces non-whitespace characters of the following fields with random letters:
  * scene:
    * title
    * details
    * url
    * code
    * director
  * image
    * title
  * gallery
    * title
    * details
  * performer
    * name
    * aliases
    * details
    * url
    * twitter
    * instagram
    * tattoos
    * piercings
  * studio
    * name
    * url
    * details
    * aliases
  * tag
    * name
    * description
    * aliases
  * movie
    * name
    * aliases
    * synopsis
    * url
    * director

### CODE REVIEW

File: pkg/sqlite/anonymise.go [hunk 1]
```diff
@@ -0,0 +1,814 @@
...
+						return fmt.Errorf("anonymising %s: %w", table.GetTable(), err)
+					}
+				}
+
+				lastID = id
+				gotSome = true
+				total++
+
+				if total%logEvery == 0 {
+					logger.Infof("Anonymised %d galleries", total)
```
Comment (bnkai) [inline pkg/sqlite/anonymise.go:476]: typo performers?

**Comment by bnkai** [comment]
Merged latest dev and did a quick test on my current db 

```bash
stash version: v0.18.0-11-g1e6324c6c (1e6324c6c - Unofficial Build) - 2022-12-02 00:50:01
INFO[2022-12-02 00:53:10] stash is listening on 127.0.0.1:9998
INFO[2022-12-02 00:53:10] stash is running at http://127.0.0.1:9998/
INFO[2022-12-02 00:53:11] New version: (87cea80e7) available.
INFO[2022-12-02 00:54:03] Anonymising folders
INFO[2022-12-02 00:54:03] Anonymising files
INFO[2022-12-02 00:54:04] Anonymising fingerprints
INFO[2022-12-02 00:54:06] Anonymised 10000 fingerprints
INFO[2022-12-02 00:54:08] Anonymised 20000 fingerprints
INFO[2022-12-02 00:54:10] Anonymised 30000 fingerprints
INFO[2022-12-02 00:54:12] Anonymised 40000 fingerprints
INFO[2022-12-02 00:54:14] Anonymised 50000 fingerprints
INFO[2022-12-02 00:54:16] Anonymised 60000 fingerprints
INFO[2022-12-02 00:54:17] Anonymised 70000 fingerprints
INFO[2022-12-02 00:54:19] Anonymised 80000 fingerprints
INFO[2022-12-02 00:54:20] Anonymised 90000 fingerprints
INFO[2022-12-02 00:54:22] Anonymised 100000 fingerprints
INFO[2022-12-02 00:54:23] Anonymising scenes
INFO[2022-12-02 00:54:33] Anonymised 10000 scenes
INFO[2022-12-02 00:54:41] Anonymised 20000 scenes
INFO[2022-12-02 00:54:48] Anonymised 30000 scenes
INFO[2022-12-02 00:54:52] Anonymising images
INFO[2022-12-02 00:54:52] Anonymised 10000 images
INFO[2022-12-02 00:54:52] Anonymised 20000 images
INFO[2022-12-02 00:54:53] Anonymised 30000 images
INFO[2022-12-02 00:54:55] Anonymised 40000 images
INFO[2022-12-02 00:54:56] Anonymised 50000 images
INFO[2022-12-02 00:54:57] Anonymised 60000 images
INFO[2022-12-02 00:54:57] Anonymising galleries
INFO[2022-12-02 00:54:57] Anonymising performers
```
The task seems to have finished according to the UI but it seems it wasnt completed ok from the lack of progress in the logs 

- a manual vacuum was needed as the anon db had the original size (5GB) even though all blobs where removed. After manually vacuuming it got down to 81MB
- scenes seem to have been anonymized ok except the `phashes` and `stash-ids`
- images and galleries seem ok
- performers, tags, studios have still all their data

**Comment by WithoutPants** [comment]
Looks like an error occurred during the performer anonymising process, and the error wasn't output and the output file was left on the disk.

**Comment by bnkai** [comment]
Yeah. For galleries/performers/studios/tags they were under 10k so no progress would be printed for them...
Log was set to debug but there were no extra prints and no errors printed 
Maybe add an  extra print with the totals anonymised after each category for double checking?

**Comment by bnkai** [comment]
error from query
```json
{
   "errors":[
      {
         "message":"querying `SELECT `performers`.`id`, `performers`.`name`, `performers`.`aliases`, `performers`.`details`, `performers`.`url`, `performers`.`twitter`, `performers`.`instagram`, `performers`.`tattoos`, `performers`.`piercings` FROM `performers` WHERE (`performers`.`id` \u003e 0) LIMIT 1000` [[]]: error executing `SELECT `performers`.`id`, `performers`.`name`, `performers`.`aliases`, `performers`.`details`, `performers`.`url`, `performers`.`twitter`, `performers`.`instagram`, `performers`.`tattoos`, `performers`.`piercings` FROM `performers` WHERE (`performers`.`id` \u003e 0) LIMIT 1000` [[]]: no such column: performers.aliases",
         "path":[
            "anonymiseDatabase"
         ]
      }
   ],
   "data":{
      "anonymiseDatabase":null
   }
}```

**Comment by bnkai** [comment]
Quick retest after latest commits
Issues mentioned above seem resolved.
The only thing missing is anonymizing or droping stash-ids from performers/studios/scenes and phashes from scenes i think

**Review by bnkai — APPROVED** [review]

**Comment by bnkai** [comment]
Tests and looks ok. Minor issue left scenes can be traced back via stashdb scraping using phashes/oshashes. Not sure if it matters and can probably be adjusted later if needed.

**Comment by dw21686** [comment]
The scene length plus the word lengths in the title/description/etc. fields is likely to be enough to determine the identity of the scene. Similarly for performers and the others.

**Pull request merged**